docs/design-doc-provider-identity-based-auth.md

```diff
@@ -0,0 +1,571 @@
+The naming convention-based or admission controller-based approaches described
+above are not straightforward to configure, especially if you also consider that
+in addition to the RBAC configurations needed to isolate the tenants
+(restricting access to the cluster-wide resources), resource quotas and network
+policies are also needed to properly isolate and fairly distribute the worker
+node resources and the network resources, respectively. Also due to the
+associated complexity, it's easy to misconfigure the cluster and difficult to
+verify a given security configuration guarantees proper isolation between
+the tenants. 
+
+As an example, consider the Kyverno `ClusterPolicy` given above:
+While the intent is to restrict the users under `tenant1` to using only the
+`ProviderConfig`s installed for them (e.g., those with names `tenant1*`), the
+scheme is broken if there exists a tenant in the system with `tenant1` as a
+prefix to its name, such as `tenant10`. 
+
+Organizations, especially with hard multi-tenancy requirements (i.e., with
+tenants assumed to be untrustworthy or actively malicious), may not prefer or
+strictly forbid such approaches. The architectural problem here, from a security
+perspective, is that the Crossplane provider (and also the core Crossplane
+components) is a shared resource itself and it requires cross-tenant privileges
+such as accessing cluster-wide resources and accessing each tenant's namespaced
+resources (especially tenant Cloud credentials). This increases the attack
+surface in the dimensions of:
+- Logical vulnerabilities (see the above example for a misconfiguration)
+- Isolation vulnerabilities: For instance, controller *workqueue*s become shared
+  resources between the tenants. How can we ensure, for instance, that the workqueue capacity
+  is fairly shared between the tenants?
+- Code vulnerabilities: As an example, consider a hypothetical Crossplane
+  provider bug in which the provider fetches another `ProviderConfig` than the
+  one declared in the managed resource, or other credentials than the ones
+  declared in the referred `ProviderConfig`. Although the logical barriers
+  enforced by the `Composition`s or the admission controllers as descibed above
+  are not broken, the too privileged provider itself breaks the cross-tenant
+  barrier.
+
+In the current Crossplane provider deployment model, when a Crossplane provider
+package is installed, there can be a single *active* `ProviderRevision`
+associated with it, which owns (via an owner reference) the Kubernetes
+deployment for running the provider. This single deployment, in turn, specifies
+a single Kubernetes service account under which the provider runs.
+
+Apart from a vulnerability perspective, there are also some other limitations to
+this architecture, which are related to identity-based authentication.
+
+**Note**: The [multi-tenancy guide][xp-mt] also mentions multi-cluster
```

Comment:
The `[multi-tenancy guide]` link is broken.

Reply:
Unfortunately can't find document link to resolve this.

Reply:
Proposed a fix for the ref `xp-mt` below.

Reply:
Fixed